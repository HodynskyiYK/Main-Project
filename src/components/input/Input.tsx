import React, { FC } from 'react'
import classnames from 'classnames'
import styles from './Input.module.scss'
import { IInput } from './inputTypes'

export const Input: FC<IInput> = (
    {
        inputClassName,
        inputName,
        inputPlaceholder,
        inputType,
        labelText,
        inputValue
    }) => {

    return (
        <>
            {
                labelText && (
                    <label
                        className={styles.Label}
                        htmlFor={inputName}
                    >{labelText}</label>
                )
            }
            <input
                className={classnames(styles.Input, inputClassName && `${styles[inputClassName]}`)}
                type={inputType}
                id={inputName}
                name={inputName}
                placeholder={inputPlaceholder}
                value={inputValue}
            />
        </>
    )
}
