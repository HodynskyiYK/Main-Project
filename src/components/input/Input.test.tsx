import React from 'react'
import { Input } from './Input'
import { render, screen } from '@testing-library/react'

describe('Input tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<Input
            inputType={'text'}
            inputName={'name'}
        />)
        expect(asFragment()).toMatchSnapshot()
    })

    test('is label in document', () => {
        render(<Input
            inputType={'text'}
            inputName={'name'}
            labelText={'label text'}
        />)
        const label = screen.getByText(/label text/i)
        expect(label).toBeInTheDocument()
    })

    test('is error message in document', () => {
        render(<Input
            inputType={'text'}
            inputName={'name'}
            errorMessage={'some error'}
        />)
        const errorMessage = screen.getByText(/some error/i)
        expect(errorMessage).toBeInTheDocument()
    })
})