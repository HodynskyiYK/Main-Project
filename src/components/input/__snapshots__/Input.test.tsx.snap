// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input tests To match snapshot 1`] = `
<DocumentFragment>
  <div
    class="inputField"
  >
    <input
      class="field"
      id="name"
      name="name"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
